src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DrawioWebView.kt

```diff
@@ -38,4 +40,15 @@ class DrawioWebView(lifetime: Lifetime) : BaseDrawioWebView(lifetime) {
         _xmlContent.set(null) // xmlLike is not xml
         send(OutgoingMessage.Event.Load(xmlLike, 1))
     }
+    fun exportSvg() : Promise<String> {
+        val result = AsyncPromise<String>()
+        send(OutgoingMessage.Request.Export(OutgoingMessage.Request.Export.XMLSVG)).then  { response ->
```

Comment:
Actually, then returns a new promise. I think you don't need `result`. But thats okay.

Reply:
hm. but what's the type of the new promise?
And how to assign the result?

Reply:
return send.then { it -> "test" + it }

Should work AFAIK. Then is like a map on the promise.

Reply:
I will give it a try for the next change 😀